src/main/java/sx/blah/discord/api/internal/json/objects/ChannelObject.java

```diff
@@ -21,60 +21,29 @@
  * Represents a json channel object.
  */
 public class ChannelObject {
-	/**
-	 * The id of the channel.
-	 */
+
 	public String id;
-	/**
-	 * The id of the guild this channel is in.
-	 */
+	public int type;
 	public String guild_id;
-	/**
-	 * The name of the channel.
-	 */
-	public String name;
-	/**
-	 * The type of the channel.
-	 */
-	public String type;
-	/**
-	 * The position of the channel.
-	 */
-	public int position;
-	/**
-	 * Whether the channel is private or not.
-	 */
-	public boolean is_private;
-	/**
-	 * Array of permission overwrites.
-	 */
+	public Integer position;
 	public OverwriteObject[] permission_overwrites;
-	/**
-	 * Topic of the channel.
-	 */
+	public String name;
 	public String topic;
-	/**
-	 * ID of the last message sent in the channel.
-	 */
 	public String last_message_id;
-	/**
-	 * When the last pin was made in the channel.
-	 */
-	public String last_pin_timestamp;
-	/**
-	 * Bitrate of the channel if it is voice type.
-	 */
-	public int bitrate;
-	/**
-	 * Maximum number of users allowed in the channel if it is voice type.
-	 */
-	public int user_limit;
-	/**
-	 * Recipients of the channel if it is private type.
-	 */
+	public Integer bitrate;
+	public Integer user_limit;
 	public UserObject[] recipients;
-	/**
-	 * Whether the channel is NSFW.
-	 */
+	public String icon;
+	public String owner_id;
+	public String application_id;
 	public boolean nsfw;
+
+	public static class Type {
+		public static final int GUILD_TEXT     = 0;
```

Comment:
I agree. Should these types be accessible to the user or placed in `internal`? I see no reason that the user should ever worry about them.